### DISCUSSION TITLE
Calling the contract prompts an error that the gas cannot be estimated

### DISCUSSION THREAD

**Discussion by yjjdick** [wevm/wagmi#1862] [Q&A]
<img width="1346" alt="3f4252a29371f9346d4fccd372c8cb0" src="https://user-images.githubusercontent.com/8224370/219121200-ce2a5c24-5958-4c6e-83e4-c964f7c1b12e.png">

Here is the method to call
`try {
        const config = await prepareWriteContract({
          address: myRouterContractAddr,
          abi: require("@/abi/fengshu.json"),
          functionName: 'fastSell',
          args: [pancakeContractAddr, _this.tokenContractAddr, lpContractChecksumAddr, exchangeAmount],
        })
        const data = await writeContract(config)
        this.successHash = data.hash
        let hash = data.hash
        const waitdata = await waitForTransaction({
          hash
        })

        if (waitdata.status == 0 || !waitdata.status) {
          _this.txStatus = 0
        } else {
          _this.txStatus = 1
        }
        if (_this.txStatus == 0) {
          $("#modal-danger").modal("show");
        } else {
          $("#modal-success").modal("show");
          _this.getTokenBalance()
        }

      } catch (e) {
        console.error(e)
        if (e.message.indexOf("User rejected request") > -1) {
          Toast.warning("操作取消")
          return
        }

        _this.trySellMsg = e.message
      }`

Please tell me how to modify this code to solve the problem of error

**Comment by jxom** [comment]
Hard to tell what is going on from that code snippet. Can you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) via a CodeSandbox or a separate repository so we can have a closer look?

**Reply by yjjdick** [reply]
> Hard to tell what is going on from that code snippet. Can you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) via a CodeSandbox or a separate repository so we can have a closer look?
This error is not inevitable, it only appears occasionally. The error roughly means that the gas limit cannot be estimated when I call the contract and needs to be set manually. Can you tell me how to manually set the gas limit when calling the contract with wagmi?

**Comment by daniserrano7** [comment]
I just had the same issue and managed to solve it by manually setting gasLimit this way:

```
const { config: presaleConfig } = usePrepareContractWrite({
  address: CONTRACT_ADDRESS,
  abi: cryptoDevsABI,
  functionName: "presaleMint",
  overrides: {
    value: ethers.utils.parseEther("0.005"),
    gasLimit: BigNumber.from(2100000),
  },
});

 const {
  data: presaleWritingData,
  isLoading: isPresaleWritingLoading,
  isSuccess: isPresaleSuccess,
  write: writePresale,
  isError: isPresaleError,
} = useContractWrite(presaleConfig);
```

Check out the `overrides` param in `usePrepareContractWrite`. Hope it helps!
